src/StackExchange.Redis/ConfigurationOptions.cs

```diff
@@ -313,7 +324,23 @@ private static bool CheckTrustedIssuer(X509Certificate2 certificateToValidate, X
             chain.ChainPolicy.UrlRetrievalTimeout = new TimeSpan(0, 0, 0);
 
             chain.ChainPolicy.ExtraStore.Add(authority);
-            return chain.Build(certificateToValidate);
+            // This only verifies that the chain is valid, but with AllowUnknownCertificateAuthority could trust
+            // self-signed or partial chained vertificates
+            var chainIsVerified = chain.Build(certificateToValidate);
+            if (chainIsVerified)
+            {
+                // Our method is "TrustIssuer", which means any intermediate cert we're being told to trust
+                // is a valid thing to trust, up until it's a root CA
+                foreach (var chainElement in chain.ChainElements)
+                {
+                    if (chainElement.Certificate.Thumbprint == authority.Thumbprint)
```

Comment:
`cert.Thumbprint` is based on SHA-1 (and can't ever be upgraded, because compat).  If you're using .NET Framework 4.8+ or .NET 5+ (or various old versions of .NET Core) you could use cert.GetCertHashString(HashAlgorithmName.SomethingBetterThanSha1).

Or, save off authority.RawData(Memory) and just SequenceEqual the element cert.RawData against the actual cert you want to compare.  Hashing isn't free, and reading Thumbprint (or calling GetCertHashString) just causes the underlying libraries to hash RawData (though with memoization).

Reply:
I'm somewhere between amused and horrified that it was previously the more correct thing, and then switched to using thumbprint with a subsequent commit.  In my best Mr. Garrison voice: "Thumbprint's bad, mm'kay?"

Reply:
This is indeed why I wanted expert eyes - switched to RawData!